src/v/storage/log_manager.cc

```diff
@@ -290,23 +290,33 @@ ss::future<> log_manager::remove(model::ntp ntp) {
 }
 
 ss::future<> log_manager::dispatch_topic_dir_deletion(ss::sstring dir) {
-    return ss::smp::submit_to(0, [dir = std::move(dir)]() mutable {
-        static thread_local mutex fs_lock;
-        return fs_lock.with([dir = std::move(dir)] {
-            return ss::file_exists(dir).then([dir](bool exists) {
-                if (!exists) {
-                    return ss::now();
-                }
-                return directory_walker::empty(std::filesystem::path(dir))
-                  .then([dir](bool empty) {
-                      if (!empty) {
-                          return ss::now();
-                      }
-                      return ss::remove_file(dir);
-                  });
-            });
+    return ss::smp::submit_to(
+             0,
+             [dir]() mutable {
+                 static thread_local mutex fs_lock;
+                 return fs_lock.with([dir] {
+                     return ss::file_exists(dir).then(
+                       [dir](bool exists) mutable {
+                           if (!exists) {
+                               return ss::now();
+                           }
+                           return directory_walker::empty(
+                                    std::filesystem::path(dir))
+                             .then([dir = std::move(dir)](bool empty) {
+                                 if (!empty) {
+                                     return ss::now();
+                                 }
+                                 return ss::remove_file(dir);
+                             });
+                       });
+                 });
+             })
+      .handle_exception_type(
+        [this, dir](const std::filesystem::filesystem_error& e) mutable {
+            vlog(
+              stlog.info, "error deleting topic directory - {}, retrying", e);
+            return dispatch_topic_dir_deletion(std::move(dir));
```

Comment:
should this be allowed to retry infinitely and without any throttle? for example, if this fiber started a directory with some extra file, since this loop doesn't clean the directory, it would be a infinite loop?

Reply:
no, it would exit in `if(!empty)`

Reply:
right, but being empty/not empty is just an example. it could be any non-transient file system issue. permissions, out of space, bad disk, etc...

Reply:
you are right, i will add a single retry